dispatcher/http.go

```diff
@@ -0,0 +1,132 @@
+package dispatcher
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"net/url"
+
+	"github.com/whosonfirst/go-webhookd/v3"
+)
+
+func init() {
+
+	ctx := context.Background()
+	err := RegisterDispatcher(ctx, "http", NewHTTPDispatcher)
+	if err != nil {
+		panic(err)
+	}
+
+	err = RegisterDispatcher(ctx, "https", NewHTTPDispatcher)
+	if err != nil {
+		panic(err)
+	}
+}
+
+// GET and POST are the only supported HTTP methods
+const GET = "GET"
+const POST = "POST"
+
+// HTTPDispatcher implements the `webhookd.WebhookDispatcher` interface for dispatching messages to a `log.Logger` instance `http.get` or `http.post`.
+type HTTPDispatcher struct {
+	webhookd.WebhookDispatcher
+	// logger is the `log.Logger` instance associated with the dispatcher.
+	logger *log.Logger
+	// url to send the message to
+	url url.URL
+	// method to use when sending the message
+	method string
+	// client to use when sending the message
+	client HTTPClient
+}
+
+// HTTPClient is an interface for `http.Client` to allow for mocking in tests.
+type HTTPClient interface {
+	Get(url string) (*http.Response, error)
+	Post(url string, contentType string, body io.Reader) (*http.Response, error)
+}
+
+// NewHTTPDispatcher returns a new `HTTPDispatcher` instance configured by 'uri' in the form of:
+//
+//	http://
+//	https://
+//
+// Messages are dispatched to the default `log.Default()` instance along with the uri parsed.
+func NewHTTPDispatcher(ctx context.Context, uri string) (webhookd.WebhookDispatcher, error) {
+	logger := log.Default()
+
+	parsed, err := url.Parse(uri)
+
+	if err != nil {
+		return nil, fmt.Errorf("Failed to parse URI, %w", err)
+	}
+
+	return NewHTTPDispatcherWithLogger(ctx, logger, *parsed, &http.Client{})
+}
+
+// NewHTTPDispatcher returns a new `HTTPDispatcher` instance that dispatches messages to `http.Get` or `http.Post`.
+func NewHTTPDispatcherWithLogger(ctx context.Context, logger *log.Logger, url url.URL, client HTTPClient) (webhookd.WebhookDispatcher, error) {
```

Comment:
Can I suggest making this something like `DispatchWithOptions` and passing in a `HTTPDispatcherOptions` struct, since it's not just about the logger but also the HTTP client.

Reply:
Changed